app/presenters/tree_builder_alert_profile_obj.rb

```diff
@@ -0,0 +1,80 @@
+class TreeBuilderAlertProfileObj < TreeBuilder
+  def initialize(name, type, sandbox, build = true, assign = nil)
+    @assign = {
+      :new     => assign[:new],
+      :current => assign[:current]
+    }
+
+    build = false unless @assign[:new][:assign_to]
+    super(name, type, sandbox, build)
+  end
+
+  def override(node, object, _pid, _options)
+    identifier = (object.name.presence || object.description)
+    if object.kind_of?(MiddlewareServer)
+      identifier += "-" + object.hostname
+    end
+
+    if @assign[:new][:assign_to].ends_with?("-tags")
+      node[:icon] = "fa fa-tag"
+    elsif @assign[:new][:assign_to] == "ext_management_system"
+      node[:image] = "svg/vendor-#{object.image_name}.svg"
```

Comment:
The problem is I dont have data to Selected Datastores option, it is reading from storage entity.  Enterprise is always empty (it is not even try to read from it as it is in the old code).  Is there any way i could know how you get the list of those possible node types?